core/ir/riscv64/decode.c

```diff
@@ -0,0 +1,247 @@
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+byte *
+decode_from_copy(void *drcontext, byte *copy_pc, byte *orig_pc, instr_t *instr)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+byte *
+decode_cti(void *drcontext, byte *pc, instr_t *instr)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+byte *
+decode_next_pc(void *dcontext, byte *pc)
+{
+    /* FIXME i#3544: Compressed instructions may be smaller */
+    return pc + RISCV64_INSTR_SIZE;
+}
+
+int
+decode_sizeof(void *drcontext, byte *pc, int *num_prefixes)
+{
+    /* FIXME i#3544: Compressed instructions may be smaller */
+    return RISCV64_INSTR_SIZE;
+}
+
+byte *
+decode_raw(dcontext_t *dcontext, byte *pc, instr_t *instr)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+bool
+decode_raw_is_jmp(dcontext_t *dcontext, byte *pc)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return false;
+}
+
+byte *
+decode_raw_jmp_target(dcontext_t *dcontext, byte *pc)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+bool
+decode_raw_is_cond_branch_zero(dcontext_t *dcontext, byte *pc)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return false;
+}
+
+byte *
+decode_raw_cond_branch_zero_target(dcontext_t *dcontext, byte *pc)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return false;
+}
+
+const instr_info_t *
+instr_info_extra_opnds(const instr_info_t *info)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+byte
+instr_info_opnd_type(const instr_info_t *info, bool src, int num)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return 0;
+}
+
+const instr_info_t *
+get_next_instr_info(const instr_info_t *info)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+byte
+decode_first_opcode_byte(int opcode)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return 0;
+}
+
+const instr_info_t *
+opcode_to_encoding_info(uint opc, dr_isa_mode_t isa_mode)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+DR_API
+const char *
+decode_opcode_name(int opcode)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return NULL;
+}
+
+opnd_size_t
+resolve_variable_size(decode_info_t *di, opnd_size_t sz, bool is_reg)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return 0;
+}
+
+bool
+optype_is_indir_reg(int optype)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return false;
+}
+
+bool
+optype_is_reg(int optype)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return false;
+}
+
+bool
+optype_is_gpr(int optype)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+    return false;
+}
+
+#ifdef DEBUG
+#    ifndef STANDALONE_DECODER
+void
+check_encode_decode_consistency(dcontext_t *dcontext, instrlist_t *ilist)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+}
+#    endif /* STANDALONE_DECODER */
+
+void
+decode_debug_checks_arch(void)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
+}
+#endif /* DEBUG */
+
+#ifdef DECODE_UNIT_TEST
+
+#    include "instr_create_shared.h"
+
+int
+main()
+{
+    bool res = true;
+    standalone_init();
+    standalone_exit();
```

Comment:
I would expect a `TODO i#3544` comment about adding some tests.

Reply:
Correct, I've added a FIXME above this code.